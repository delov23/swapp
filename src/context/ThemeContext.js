import { createContext } from 'react';

export const LIGHT_THEME = 'light';
export const DARK_THEME = 'dark';

const ThemeContext = createContext(LIGHT_THEME);

export default ThemeContext;
