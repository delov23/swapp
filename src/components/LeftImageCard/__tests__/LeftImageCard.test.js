import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import { LeftImageCard } from '../';
import ThemeContext, {
  LIGHT_THEME,
  DARK_THEME,
} from '../../../context/ThemeContext';

const EXAMPLE_PROPS = {
  image: 'https://img.com/img.png',
  text: 'Example Example',
  href: '/logout',
  width: '50vw',
};

describe('<Button />', () => {
  it('should match snapshot', () => {
    expect(shallow(<LeftImageCard {...EXAMPLE_PROPS} />)).toMatchSnapshot();
  });

  it('should support LIGHT_THEME', () => {
    const wrapper = mount(
      <ThemeContext.Provider value={LIGHT_THEME}>
        <Router>
          <LeftImageCard {...EXAMPLE_PROPS} />
        </Router>
      </ThemeContext.Provider>,
    );

    expect(wrapper.find(LeftImageCard).find('a')).toHaveClassName('cardLight');
  });

  it('should support DARK_THEME', () => {
    const wrapper = mount(
      <ThemeContext.Provider value={DARK_THEME}>
        <Router>
          <LeftImageCard {...EXAMPLE_PROPS} />
        </Router>
      </ThemeContext.Provider>,
    );

    expect(wrapper.find(LeftImageCard).find('a')).toHaveClassName('cardDark');
  });

  it('should display props from the parent', () => {
    const wrapper = mount(
      <Router>
        <LeftImageCard {...EXAMPLE_PROPS} />
      </Router>,
    );

    expect(wrapper.find(LeftImageCard)).toHaveProp(
      'image',
      EXAMPLE_PROPS.image,
    );
    expect(wrapper.find(LeftImageCard)).toHaveProp('text', EXAMPLE_PROPS.text);
    expect(wrapper.find(LeftImageCard).find('a')).toHaveProp(
      'href',
      EXAMPLE_PROPS.href,
    );
    expect(wrapper.find(LeftImageCard).find('a')).toHaveStyle(
      'width',
      EXAMPLE_PROPS.width,
    );
  });

  it('should display props from the parent', () => {
    const wrapper = mount(
      <Router>
        <LeftImageCard {...{ ...EXAMPLE_PROPS, image: '' }} />
      </Router>,
    );

    expect(wrapper.find('img')).toHaveProp(
      'src',
      'https://wolper.com.au/wp-content/uploads/2017/10/image-placeholder.jpg',
    );
  });
});
