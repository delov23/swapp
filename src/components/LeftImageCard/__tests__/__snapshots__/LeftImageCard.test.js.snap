// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<Link
  className="cardLight"
  style={
    Object {
      "width": "50vw",
    }
  }
  to="/logout"
>
  <img
    alt=""
    className="image"
    src="https://img.com/img.png"
  />
  <h3
    className="heading"
  >
    Example Example
  </h3>
</Link>
`;
